examples/pruning/tensorflow_integration.py

```diff
@@ -0,0 +1,162 @@
+"""
+Optuna example that demonstrates a pruner for Tensorflow.
+
+In this example, we optimize the hyperparameters of a neural network for hand-written
+digit recognition in terms of validation accuracy. The network is implemented by Tensorflow and
+evaluated by MNIST dataset. Throughout the training of neural networks, a pruner observes
+intermediate results and stops unpromising trials.
+
+You can run this example as follows:
+    $ python tensorflow_integration.py
+
+"""
+
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import shutil
+import tensorflow as tf
+
+import optuna
+
+MODEL_DIR = "/tmp/mnist_convnet_model"
```

Comment:
I basically agree with your opinion.
However, I have a concern. Since `tempfile.mkdtemp` does not clean up the created directory, if users execute this example many times, it will consume a huge capacity of their disk (over 700MB per execution).
Should we accept it? Or should we introduce a clean up mechanism?

Reply:
As a result of offline discussion, we decided to add `shutil.rmtree()` after `study.optimization()`.

Reply:
Done: https://github.com/pfnet/optuna/pull/323/commits/e84c158e254c974e7a4152d006b3558d50f72eac